src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -44,57 +50,190 @@ export class PermissionsTableComponent implements OnInit {
+    private usersService: UserService,
+    private groupService: GroupService) {
   }
 
   public ngOnInit() {
+    this.setPermissionsForResource(this.resourceType);
+
     this.roles = ROLES[this.resourceType];
-    this.entities = this.getEntities(this.entityType);
-    this.otherEntities = this.getOtherEntities(this.entityType);
+
+    this.rolesCheckbox = {};
+    this.roles.forEach(r => {
+      this.rolesCheckbox[r] = false;
+    });
   }
 
-  private getEntities(entityType: string): Entity[] {
-    switch (entityType) {
-      case 'users':
-        // TODO get from rolesService instead
-        return [
-          {name: 'marvec', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'livthomas', roles: ['read', 'share', 'write']}
-        ];
-      case 'groups':
-        // TODO get from rolesService instead
-        return [
-          {name: 'managers', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'employees', roles: ['read', 'share']},
-          {name: 'contractors', roles: ['read']}
-        ];
-      default:
-        throw Error('unknown entity type');
+  private getEntities(entityType: string, permissions: Permissions): Permission[] {
+    let entities = permissions[entityType];
+    if (entities) {
+      return entities;
+    } else {
+      throw Error('unknown entity type');
+    }
+  }
+
+  private setPermissionsForResource(resourceType: string) {
+    let parent = this;
+    let getPermissionsFunctions = {
+      'organization': function() {return parent.organizationService.getPermissions();} ,
+      'project': function() {return parent.projectService.getPermissions();},
+      'collection': null, // TODO
+      'view': null // TODO
+    };
+    let getPermissions = getPermissionsFunctions[resourceType];
+    if (getPermissions) {
+      getPermissions().subscribe((permissions: Permissions) => {
+          this.entities = this.getEntities(this.entityType, permissions);
+          this.getOtherEntities(this.entityType);
+        });
+    } else {
+      throw Error('unknown resorce type');
     }
   }
 
-  private getOtherEntities(entityType: string): string[] {
+  private getOtherEntities(entityType: string) {
     switch (entityType) {
       case 'users':
-        // TODO get from usersService instead
-        return ['alicak', 'kubedo', 'jkotrady', 'kulexpipiens'];
+        this.usersService.getUsers().subscribe(
+          (userEntities: string[]) => {
+            for (let entity of this.entities) {
+              let index = userEntities.indexOf(entity['name']);
+              if (index >= 0) {
+                userEntities.splice(index, 1);
+              }
+            }
+            this.otherEntities = userEntities;
+          }
+        );
+        break;
       case 'groups':
-        // TODO get from groupsService instead
-        return ['directors', 'customers'];
+        this.groupService.getGroups().subscribe(
+          (groupEntities: string[]) => {
```

Comment:
There is exactly the same block of code for users and groups here. It can be extract to a new method so we do not have unnecessary duplicates here.